src/lib/tests/test_kernels.cpp

```diff
@@ -0,0 +1,86 @@
+#include "gtest/gtest.h"
+#include <cassert>
+#include <iostream>
+#include <random>
+
+#include "arrops.h"  //  fancy approaches
```

Comment:
One day in the future we'll decide what we want our public API to be, and it won't involve names like "arrops.h" :p

Reply:
ack